code/game/objects/items.dm

```diff
@@ -1010,3 +1010,159 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
+
+		if(custom_materials[SSmaterials.GetMaterialRef(/datum/material/adamantine)] || custom_materials[SSmaterials.GetMaterialRef(/datum/material/runite)])
+			M.apply_damage(20, BRUTE, BODY_ZONE_HEAD, wound_bonus = 10) //oof my tooth
+
+		else if(custom_materials[SSmaterials.GetMaterialRef(/datum/material/diamond)] || custom_materials[SSmaterials.GetMaterialRef(/datum/material/titanium)])
+			M.apply_damage(15, BRUTE, BODY_ZONE_HEAD, wound_bonus = 7)
+
+		else if(custom_materials[SSmaterials.GetMaterialRef(/datum/material/iron)])
+			M.apply_damage(10, BRUTE, BODY_ZONE_HEAD, wound_bonus = 5) 
+
+		else if(custom_materials[SSmaterials.GetMaterialRef(/datum/material/wood)])
+			M.apply_damage(5, BRUTE, BODY_ZONE_HEAD)
+			if(S && S.tastes && S.tastes.len)
+				S.tastes += "wood chips and sawdust"
+				S.tastes["wood chips and sawdust"] = 3
+
+		else if(custom_materials[SSmaterials.GetMaterialRef(/datum/material/glass)])
+			M.apply_damage(10, BRUTE, BODY_ZONE_HEAD, wound_bonus = 5, sharpness = TRUE) //cronch
+			//biting glass doesn't mean you swallow the glass, but it DOES leave the shard in there if you keep eating like an idiot
+			/// The glass shard that is spawned into the source item
+			var/obj/item/shard/broken_glass = null
+
+			if(istype(src, /obj/item/reagent_containers)) //if it's a glass reagent container...
+				if(reagents.reagent_list && reagents.reagent_list.len)
+					for(var/P in reagents.reagent_list)
+						if(istype(P, /datum/reagent/toxin/plasma) || istype(P, /datum/reagent/stable_plasma))
+							broken_glass = new /obj/item/shard/plasma(src) //if there's plasma in the beaker, make a plasma glass shard
+							break
+			
+			if(!broken_glass)
+				if(found_mats && custom_materials[SSmaterials.GetMaterialRef(/datum/material/plasma)])
+					broken_glass = new /obj/item/shard/plasma(src) //or if it's plasma glass, make a plasma glass shard
+				else
+					broken_glass = new /obj/item/shard(src) //and if it's normal glass, make a normal shard
+
+			playsound(M, "shatter", 25, TRUE)
+			if(source_item)
+				source_item.contents += broken_glass //puts the glass back into the source item...
+				source_item.contents -= src
+			else
+				broken_glass.on_accidental_consumption(M, user)
+
+			qdel(src)
+
+		else if(custom_materials[SSmaterials.GetMaterialRef(/datum/material/plastic)])
+			M.adjust_disgust(17) //(33 +  17 = 50) finding metal is pretty disgusting, but finding plastic is always the worst
+		
+		M.adjust_disgust(33)
+		M.visible_message("<span class='warning'>[M] looks like [M.p_theyve()] just bitten into something hard.</span>", \
+						"<span class='warning'>Eugh! Did I just bite into something?</span>")
+
+	else if(w_class == WEIGHT_CLASS_TINY) //small items like soap or toys that don't have mat datums
+		/// M's chest (for cavity implanting the item)
+		var/obj/item/bodypart/chest/CV = M.get_bodypart(BODY_ZONE_CHEST)
+		if(CV.cavity_item)

```

Comment:
holy shit this is amazing

Reply:
oh, wait, it doesn't cycle out the item, it just doesn't let this new item go in if you already have an item inside of you

still cool, but not quite as good as I thought